### PR TITLE
All the things

### PR DESCRIPTION
**Pull request by thamaranth**
it does all the things. If you want to test the date npm run why, everything else npm test.

### CODE REVIEW

File: src/test/checkTest.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import chai, { expect } from 'chai'
+
+
+  describe('check', () => {
+
+    it('does the thing', () => {
+    })
+
+
+  })
```
Comment (DianaVashti) [inline src/test/checkTest.js]: What is this file? The filename does not lead to anything else committed and has no logic yet. Omit from PRs.

File: src/test/dateTest.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import chai, { expect } from 'chai'
+import times from '../times'
+import date from '../main/date'
+import moment from 'moment'
+
+  describe('date', () => {
+
+    it('does the thing', () => {
+    })
```
Comment (DianaVashti) [inline src/test/dateTest.js]: Omit placeholding code (empty params + empty function) from PRs

File: src/test/dateTest.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+    describe('date', () => {
+      it('changes the given dates to yyyy-mm-dd format', () => {
+        console.log(date(times))
+        // expect(date(times)).to.equal('1972-11')
```
Comment (DianaVashti) [inline src/test/dateTest.js]: Why is the expect commented out?

File: src/test/disemvowelTest.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
+import chai, { expect } from 'chai'
+import disemvowel from '../main/disemvowel'
+
+
+  describe('disemvowel', () => {
+
+    it('does the thing', () => {
+    })
```
Comment (DianaVashti) [inline src/test/disemvowelTest.js]: ?

File: src/test/disemvowelTest.js [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+
+    describe('disemvowel', () => {
+      it('removes the vowels from a given string', () => {
+
+        expect(disemvowel('yo check this out'))
+        .to.equal('ychckthst')
+        expect(disemvowel('yaldcakhcvasdnskcbjahvLZKxavkaacauceyficksgukUFugsjnnnxnnxFSHDBACKUgeuagfaukfavheywftcgjvcepmjgacd khufd'))
+        .to.equal('yldckhcvsdnskcbjhvLZKxvkccyfcksgkFgsjnnnxnnxFSHDBCKggfkfvhywftcgjvcpmjgcdkhfd')
+        expect(disemvowel('yo    check this out'))
+        .to.equal('ychckthst')
```
Comment (DianaVashti) [inline src/test/disemvowelTest.js]: each of these tests should have their own `it('what it is expectuing'...)` For example the second seems to be testing for your functions ability to handle a really long input so you should say that i.e. `it('Removes all vowels regardless of string size', ...)`

File: src/main/date.js [hunk 6]
```diff
@@ -0,0 +1,30 @@
+import moment from 'moment'
+import times from '../times'
+
+function date(arr) {
+  let i = 0
+  const regex = /(\w+)[\#](\w+)[\#](\w+)/ig
+  const regexStar = /(\w+)\*(\w+)\*(\w+)/ig
+  while ( i < 1000) {
+    let starKiller = arr[i].replace(regexStar,'$3 $2 $1')
```
Comment (DianaVashti) [inline src/main/date.js:9]: Does this convert asterisks to dashs? Could use a more precise var name

File: src/main/date.js [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+    let pounds = arr[i].replace(regex,'$2 $1 $3' )
+    let badPounds = arr[i].replace(regex,'20$2 $1 $3')
```
Comment (DianaVashti) [inline src/main/date.js:11]: same for these two. The var name does not give me a good idea of what they do.

File: src/extras/shuffleFactorial.js [hunk 8]
```diff
@@ -0,0 +1,97 @@
...
+//     if ( finalArr.includes(strang) === true ){
+//       return shuffleFactorial(strang.shuffle())
+//     } else if (finalArr.includes(strang) === false ){
+//       finalArr.push(strang)
+//       return shuffleFactorial(strang.shuffle())
+//     }
+//   }
+// }
+//
+// console.log(shuffleFactorial('dawgo').length)
```
Comment (mistakeets) [inline src/extras/shuffleFactorial.js:97]: not sure what this commented out code is for. If you want to keep this code for future reference, I would recommend taking this out of the repo, and copy / paste the code to a new file outside of this repo.

File: src/extras/totalPermutations.js [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+  } return secondEmptyArray
+}
+
+const totalPermutationsDenominator = function(argh){
+  return argh.reduce(function(a,b){return a*b})
+}
+
+export default function totalPermutations(stringy){
+  return factorial(stringy.length) / totalPermutationsDenominator(innerArrayFactorial(tallyRepeats(stringy)))
+}
```
Comment (mistakeets) [inline src/extras/totalPermutations.js:48]: looks good! clean / easy to read code

File: src/main/disemvowel.js [hunk 10]
```diff
@@ -0,0 +1,7 @@
+const disemvowel = str => {
+  return str.replace(/[aeiou]/ig,'').replace(/\s+/ig,'')
```
Comment (mistakeets) [inline src/main/disemvowel.js:2]: you could also place the \s inside the vowels regex statement if you want to. something like 

str.replace(/[\saeiou]/ig,'')

File: src/main/guess.js [hunk 11]
```diff
@@ -0,0 +1,28 @@
+const readline = require('readline');
```
Comment (mistakeets) [inline src/main/guess.js:1]: good work on the guess a number program!

File: src/main/scrab.js [hunk 12]
```diff
@@ -0,0 +1,80 @@
...
+    // if the 9 doesn't exist in objToo
+      //let blah = letter.keys
+      // ojbToo.letter = [blah]
+    //else push letter to the existing num (example c)
+
+    // for (let keyToo in obj){
+    //   if (obj[key] == obj[keyToo]){
+    //   objToo = obj[keyToo] + ':' + key
+  //     }
+  //   }
```
Comment (mistakeets) [inline src/main/scrab.js:60]: for readability I would recommend taking this out and saving it for future reference somewhere outside of the git repo.

**Review by mistakeets — COMMENTED** [review]
Good work!

**Pull request merged**